lib/api/controllers/funnelController.js

```diff
@@ -443,8 +443,7 @@ class FunnelController {
             }
 
             return Bluebird.reject(modifiedRequest.error);
-          })
-          .catch(() => Bluebird.reject(_error));
+          });
```

Comment:
I disagree: we _have_ to enforce an error inheriting the `KuzzleError` abstract class, if only to make sure that the expected fields are there, and preventing reading possible undefined values.

To fix the problem mentionned in the PR, I would instead do this:

```js
          .catch(error => {
            if (!(error instanceof Error) || !(error instanceof KuzzleError)) {
              throw new PluginImplentationError(error);
            } else {
              throw error;
            }
          }
```

Also, PR fixing bugs should as often as possible include new unit tests reproducing the problem, with the PR fixing those. And in that specific case, I believe that this bug does warrant new unit tests, since we're fixing a behavior directly visible to plugin developers.

Reply:
They can't set another error than a KuzzleError on a request because the `setError` method wrap any other error into an `InternalError` and you can't manually set the error: https://github.com/kuzzleio/kuzzle-common-objects/blob/master/lib/request.js#L232
I still kept your proposal and I added tests.

Reply:
That's true for `setError`, but since we're in a catched promise, we have to assume that the resulting error comes from a `throw` (or a `reject`), and that `setError` hasn't necessarily been called